File: 23.go
package main

import(
	"fmt"
)

func isAbundant(natural int) bool {
	var counter int = 1

	for i := 2; i < natural; i++ {
		if natural % i == 0 {
			counter += i
		}
	}


	return counter > natural
}

func main (){
	var abundants[]int
	var limit int = 28123
	var counter int = 0
	var isSumOfTwoAbundants[28124]bool

	for i := 2; i < limit; i++{
		if isAbundant(i) {
			abundants = append(abundants, i)
		}
	}

	for j := 0; j < len(abundants); j++ {
		for k := 0; k < len(abundants); k++ {
			if abundants[k] + abundants[j] <= limit {
				isSumOfTwoAbundants[abundants[j] + abundants[k]] = true
			} else {
				break
			}
		}
	}

	for l := 0; l < len(isSumOfTwoAbundants); l++ {
		if !isSumOfTwoAbundants[l] {
			counter += l
		}
	}
	fmt.Println(counter)	
}

//prints 4179871

File: 23.py
def is_abundant (integer):
    final = 0
    for i in range(1, integer):
        if integer % i == 0:
            final += i
    return final > integer

sum_of_two_abundants = [False] * 28124
limit = 28123
the_abundants = [i for i in range(0, limit + 1) if is_abundant(i)]
counter = 0

for i in the_abundants:
    for j in the_abundants:
        if i + j <= limit:
            sum_of_two_abundants[i + j] = True


for k in range(0, len(sum_of_two_abundants)):
    if sum_of_two_abundants[k] == False:
        counter += k

print(counter)
#prints 4179871


